### PR TITLE
SALTO-1035 added objectMapping to missing fields list

### PR DESCRIPTION
**Pull request by nitzan-haim**

### CODE REVIEW

**Comment by nitzan-haim** [comment]
General note: the `ObjectMapping` type is borrowed from `DuplicateRuleMatchRule`. It solves this bug and fixes objectMapping's fields representation, but I can’t predict other consequences. The reason I did that here is because the field's structure is identical. (see https://salto-io.atlassian.net/browse/SALTO-1035 for more details)

File: packages/salesforce-adapter/src/filters/missing_fields.json [hunk 1]
```diff
@@ -294,16 +294,8 @@
...
-      },
-      {
-        "name": "outputField",
-        "type": "STRING"
-      },
-      {
-        "name": "outputObject",
-        "type": "STRING"
+        "name": "objectMapping",
+        "type": "ObjectMapping"
```
Comment (ori-moisis) [inline packages/salesforce-adapter/src/filters/missing_fields.json:298]: from the documentation page it seems like this is actually a list. not sure if we have a way to represent that in this definition or if we can only do that in the hardcoded lists json.

**Review by ori-moisis — APPROVED** [review]
assuming you tested fetch and deploy, this looks like the right change to make

File: packages/salesforce-adapter/src/filters/missing_fields.json [hunk 2]
```diff
@@ -294,16 +294,8 @@
...
       {
-        "name": "inputField",
-        "type": "STRING"
-      },
-      {
-        "name": "outputField",
-        "type": "STRING"
-      },
-      {
-        "name": "outputObject",
```
Comment (omrilit) [inline packages/salesforce-adapter/src/filters/missing_fields.json:305]: From reading the documentation it seems like this field is still required in for `LeadConvertSettings`. Why did you delete it? Am I missing something? Please see my comment at https://salto-io.atlassian.net/browse/SALTO-1035

File: packages/salesforce-adapter/src/filters/missing_fields.json (hunk 1 above)
Comment (omrilit) [reply packages/salesforce-adapter/src/filters/missing_fields.json:298]: Seems like this is a set, which means we should mark it as a list and take care for its order

File: packages/salesforce-adapter/src/filters/missing_fields.json (hunk 2 above)
Comment (ori-moisis) [reply packages/salesforce-adapter/src/filters/missing_fields.json:305]: pretty sure the documentation is wrong, both the example at the bottom of the page and the metadata WSDL seem to show this field is nested inside the `objectMapping`

relevant part of the WSDL:
```xml
<xsd:complexType name="LeadConvertSettings">
  <xsd:complexContent>
    <xsd:extension base="tns:Metadata">
      <xsd:sequence>
        <xsd:element name="allowOwnerChange" minOccurs="0" type="xsd:boolean"/>
        <xsd:element name="objectMapping" minOccurs="0" maxOccurs="unbounded" type="tns:ObjectMapping"/>
        <xsd:element name="opportunityCreationOptions" minOccurs="0" type="tns:VisibleOrRequired"/>
      </xsd:sequence>
    </xsd:extension>
  </xsd:complexContent>
</xsd:complexType>
```

Comment (omrilit) [reply packages/salesforce-adapter/src/filters/missing_fields.json:305]: I think that you are right, thanks!

File: packages/salesforce-adapter/src/filters/missing_fields.json (hunk 1 above)
Comment (nitzan-haim) [reply packages/salesforce-adapter/src/filters/missing_fields.json:298]: We already marked this as a list a while ago: https://github.com/salto-io/salto/blob/c3006720f66bc23b1cf545905b075e9d0a5a8da5/packages/salesforce-adapter/src/filters/hardcoded_lists.json#L103

**Review by omrilit — APPROVED** [review]
